### DISCUSSION TITLE
Re-render after validation

### DISCUSSION THREAD

**Discussion by pandika-csgi** [clientIO/joint#1962] [Q&A] [answered]
I have a case when `inspector` have an content-editable input connecting to `cell` then if blank will return error like this 
![image](https://user-images.githubusercontent.com/106044262/209309635-69f1fce2-462e-43af-811b-5484afad3b44.png)

In my case:
1. `Input` initiated with empty string, so will show an error message
2. When we fill out the input, error will removed then text on the `cell` will same as the user input
3. User delete the text on `input`, the text cell got updated but error not showing again, it might be because this code on `validateInputField` function

```
    if (path === 'InputName' && !value) {
      const cellTitle = inspector.options.cell;
      const cellConfiguration = cellTitle.get('cellTitle');
 
      cellTitle.set('cellTitle', { ...cellConfiguration, name: '' });
    }
```

My question is, how to run `updateCell` with specific input we already defined before on inspector, I dont know how to get inspector element, so I can use it in `updateCell`

**Comment by kumilingus** [comment, marked as answer]
To update only a specific field/attribute you can do:
```js
const path = 'attrs/body/stroke';
inspector.updateCell(inspector.el.querySelector(`[data-attribute="${path}"]`), path);
```

**Reply by pandika-csgi** [reply]
Thanks but, if I run that inside `validateInputField` it will be causing error since `validateInputField` called here and then run `updateCell()`, any idea? 
 ```
ui.Inspector.create('.inspector-container', {
        cell,
        groups,
        inputs,
        renderFieldContent: this.createCustomFieldContent,
        ...
        multiOpenGroups: false,
        validateInput: this.validateInputField,
      }).updateCell();
 ```

**Reply by kumilingus** [reply]
What error do have in mind?

You can call `updateCell()` from the `validateInput` method. It receives the `inspector` instance as the last parameter.
```js
ui.Inspector.create('.inspector-container', {
   /* ... */
   validateInput: function(element, path, type, inspector) {
       // validation
      const path = 'attrs/body/stroke';
      inspector.updateCell(inspector.el.querySelector(`[data-attribute="${path}"]`), path);
   }
 });
```

**Reply by pandika-csgi** [reply]
ERROR RangeError: Maximum call stack size exceeded
I think it is because at the end of creating Inspector I called `updateCell()` too, so I have two updateCell after creating Inspector and inside validateInput

**Reply by kumilingus** [reply]
The `updateCell()` at the end is called just after the inspecto initialization.
The `updateCell()` inside the `validateInput()` runs everytime the user changes an input.

The `updateCell()` method takes the inspector current state (e.g. value of `<input/>) and saves it to the model.

I'm not sure what you're actually trying to solve. Could you post some code here that can be run? Could you please consult your answers with the translation software please (sometimes it's hard to understand).

**Reply by pandika-csgi** [reply]
Sorry for my bad english, right now the case is closed, thank you for your response
